config/database.yml

```diff
@@ -1,6 +1,8 @@
 development:
   adapter: postgresql
-  database: discourse_development
+  database: <%= ENV['DISCOURSE_DB_NAME'] || 'discourse_development' %>
```

Comment:
Sorry, I'm not following. In production, database.yml is ignored and only environment is used. I thought you were suggesting to do the same for development/test. Did I misunderstand?